src/realm/sync/instruction_applier.cpp

```diff
@@ -1,30 +1,58 @@
 #include <realm/sync/instruction_applier.hpp>
 #include <realm/sync/object.hpp>
 #include <realm/set.hpp>
+#include <realm/util/scope_exit.hpp>
 
 #include <realm/group.hpp>
 
 namespace realm::sync {
 
-namespace {
-REALM_NORETURN void throw_bad_transaction_log(std::string msg)
+REALM_NORETURN void InstructionApplier::bad_transaction_log(const std::string& msg) const
 {
-    throw BadChangesetError{std::move(msg)};
-}
-REALM_NORETURN void bad_transaction_log(const char* msg)
-{
-    throw_bad_transaction_log(msg);
```

Comment:
It's probably not a big difference here since it's only going up to four throw locations, but switching away from throwing in only one spot is going to make the compiled size a bit bigger.